### PR TITLE
Add min date for milestone 

### LINKED ISSUE #8926: [BUG] need to add some control to the Milestone date field
### Description

While adding milestones  I accidentally wrote in the year part of the date field "0008"; the form accepted the value showing a mileston in year 8. We should probably add some kind of control, like a minimum date

### Screenshots

Screenshot 2023-09-02 202609
Screenshot 2023-09-02 202624

### Do you want to work on this issue?

Yes

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Giuliano1993**
## Fixes Issue

Closes #8926 

## Changes proposed

 -  added a front-end control in input field as min date
 -  added a backend control in milestone schema with min date

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers

I added as min date 01-01-1950 as it seemed a reasonable limit to me, but can be changed if is not good enough

### CODE REVIEW

**Review by accodes21 — COMMENTED** [review]
Looks good to me.

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for the contribution 👍 

Why `1950` as the min date? Maybe 1970 is better?

File: models/Profile/Milestone.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ const MilestoneSchema = new Schema({
   date: {
     type: Date,
     required: true,
+    min: '1970-01-01'
```
Comment (eddiejaoude) [inline models/Profile/Milestone.js:13]: I could see this feature in the official documentation?

Comment (Giuliano1993) [reply models/Profile/Milestone.js:13]: BTW if you meen in the official documentation, here's the link @eddiejaoude 
https://mongoosejs.com/docs/tutorials/dates.html#validators

Comment (eddiejaoude) [reply models/Profile/Milestone.js:13]: oh yes, I couldn't see the `min` for dates in the docs, thank you 👍

**Review by eddiejaoude — APPROVED** [review]
Great! Thank you 👍

**Pull request merged**